src/frontends/tensorflow_common/src/op/reverse_sequence.cpp

```diff
@@ -14,13 +19,47 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 OutputVector translate_reverse_sequence_op(const NodeContext& node) {
-    default_op_checks(node, 2, {"ReverseSequence"});
+    default_op_checks(node, 2, {"ReverseSequence"}, true);
     auto input = node.get_input(0);
     auto seq_lengths = node.get_input(1);
 
     // retrieve attributes
     auto seq_dim = node.get_attribute<int64_t>("seq_dim");
     auto batch_dim = node.get_attribute<int64_t>("batch_dim", 0);
+    // Handling negative values by adding rank if needed
+    if (seq_dim < 0) {
+        seq_dim += input.get_shape().size();
+    }
+    if (batch_dim < 0) {
+        batch_dim += input.get_shape().size();
+    }
+
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
+    if (complex_type_mark) {
+        auto const_one = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+        auto seq_dim_tensor = make_shared<v1::Select>(
+            make_shared<v1::Equal>(make_shared<v1::Rank>(input), const_one),
+            make_shared<v0::Constant>(element::i64, Shape{}, seq_dim),
+            make_shared<v1::Select>(make_shared<v1::Equal>(make_shared<v1::Rank>(input), const_one),
+                                    make_shared<v1::Subtract>(make_shared<v1::Rank>(input), const_one),
+                                    make_shared<v0::Constant>(element::i64, Shape{}, -1)));
```

Comment:
not clear logic.
please create helper function to update `seq_dim` and `batch_dim` because the logic is the same.
please create separate variable for each step like `auto rank =...` and `auto minus_one_const=...` etc. This way we can clarify what the logic is meant here.